core/src/main/java/org/elasticsearch/index/shard/ShardStateMetaData.java

```diff
@@ -100,6 +110,9 @@ public void toXContent(XContentBuilder builder, ShardStateMetaData shardStateMet
             builder.field(VERSION_KEY, shardStateMetaData.version);
             builder.field(PRIMARY_KEY, shardStateMetaData.primary);
             builder.field(INDEX_UUID_KEY, shardStateMetaData.indexUUID);
+            if (shardStateMetaData.allocationId != null) {
```

Comment:
probably when we read legacy allocationId. Not sure if we need to support the re-serialization of those. When can it happen?